### ISSUE TITLE
Completed Pods should not block the updates 

### ISSUE THREAD

**Issue by szuecs** [zalando-incubator/cluster-lifecycle-manager#50]
Logs showing PDB violations:
```
time="2018-06-05T13:48:25Z" level=info msg="Pod Disruption Budget violated" cluster=stups node=ip-172-31-17-54.eu-central-1.compute.internal ns=default pod=b-hdq3tk6ivnwstiy22f2zfejwg worker=2
time="2018-06-05T13:48:25Z" level=info msg="Pod Disruption Budget violated" cluster=stups node=ip-172-31-17-54.eu-central-1.compute.internal ns=default pod=b-htbbbaynjiw53xocviwpjmbvs worker=2
time="2018-06-05T13:48:25Z" level=info msg="Pod Disruption Budget violated" cluster=stups node=ip-172-31-17-54.eu-central-1.compute.internal ns=default pod=b-qybqt9bv8hmqe1chnef3smsv3 worker=2
```

Pods are already completed:
```
% kubectl get pods b-hdq3tk6ivnwstiy22f2zfejwg b-htbbbaynjiw53xocviwpjmbvs b-qybqt9bv8hmqe1chnef3smsv3
NAME                          READY     STATUS      RESTARTS   AGE
b-hdq3tk6ivnwstiy22f2zfejwg   0/1       Completed   0          1h
b-htbbbaynjiw53xocviwpjmbvs   0/1       Completed   0          1h
b-qybqt9bv8hmqe1chnef3smsv3   0/1       Completed   0          1h
```

PDB that stops the rolling update:
```
% kubectl get pdb cdp-controller-builder
NAME                     MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
cdp-controller-builder   N/A             0                 0                     13d
```

Workaround:
```
% kubectl get pods | awk '/Completed/ {print $1}' | xargs kubectl delete pod
```

**Comment by aermakov-zalando** [comment]
Fixed in #51.

**Comment by aermakov-zalando** [comment]
The reason is that PDBs are checked during eviction before pod status is checked, so if a matching PDB doesn't allow to evict a pod Kubernetes won't even check that the pod is already terminated.
